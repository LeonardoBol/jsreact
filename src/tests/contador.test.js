import Contador from '../contador'
//import { render } from '@testing-library/react'
//import '@testing-library/jest-dom/extend-expect'
import { shallow } from 'enzyme'



describe('Pruebas sobre el componente contador',()=>{

   


   test('Probando con enzyme', ()=>{

    const wrapper = shallow(<Contador propNum={0} />)
    expect(wrapper).toMatchSnapshot()

})



test('Traer button',() => {
    const wrapper = shallow(<Contador propNum={10} />)
    const btn1 = wrapper.find('button').at(0)
    console.log(btn1.html())
  })


  test('Simular click boton suma',() => {
    const wrapper = shallow(<Contador propNum={10} />)
    const btn1 = wrapper.find('button').at(0).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toBe('11')
  
  })
  
  


})