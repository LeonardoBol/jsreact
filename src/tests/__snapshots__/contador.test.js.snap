// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente contador Probando con enzyme 1`] = `
<Fragment>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    ¡Más!
  </button>
  <button
    onClick={[Function]}
  >
    ¡Menos!
  </button>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
     ¡Reset!
  </button>
</Fragment>
`;
